targets/TARGET_Realtek/TARGET_AMEBA/RTWInterface.cpp

```diff
@@ -191,85 +179,93 @@ nsapi_error_t RTWInterface::connect()
             return NSAPI_ERROR_PARAMETER;
     }
 
+    if (_channel > 0 && _channel < 14) {
+        uint8_t pscan_config = PSCAN_ENABLE;
+        wifi_set_pscan_chan(&_channel, &pscan_config, 1);
+    }
+
     ret = wifi_connect(_ssid, sec, _pass, strlen(_ssid), strlen(_pass), 0, (void *)NULL);
     if (ret != RTW_SUCCESS) {
         printf("failed: %d\r\n", ret);
         return NSAPI_ERROR_NO_CONNECTION;
     }
```

Comment:
As Mika mentioned, you will still need a "link up" callback here after the "wifi_connect", to ensure the upcall occurs if this is the second connect, and the stack is already bound to you. That's the direct mirror of the "link down" callback from "disconnect".

The callback in "power_up" then handles the case of the first connect, when the stack wasn't attached to hear this link-up callback.

Reply:
Thanks. Will add rtw_emac.wlan_emac_link_change(true);afterwifi_connect .